documentation/openapi-authoring-automated-guidelines.md

```diff
@@ -351,4 +353,258 @@ Control characters are not allowed in a specification.
+```
+Certificate_Activate
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2055" />R2055 OneUnderscoreInOperationId
+**Output Message**: Only 1 underscore is permitted in the operation id, following Noun_Verb conventions.
+
+**Description**: An operationId can have exaclty one underscore, not adhering to it can cause errors in code generation.
+
+**Why the rule is important**: Given an operationId of the form `Noun_Verb`, AutoRest breaks the operation id into its `Noun` and `Verb` where `Noun` becomes name of the operations class and the `Verb` becomes the name of the method in that class. Not adhering to this format can cause AutoRest to fail during code generation.
+
+**How to fix the violation**: Ensure operationId is of the form `Noun_Verb` and contains exactly one underscore.
+
+**Bad Example**:
+```
+Activate_Primary_Certificate
+```
+**Good Example**:
+```
+PrimaryCertificate_Activate
+```
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R3023" />R3023 OperationsAPIImplementation
+**Output Message**: Operations API must be implemented for '{0}'.
+
+**Description**: Per ARM guidelines, each RP must expose an operations API that returns information about all the operations available with the service.
+
+**Why the rule is important**: For better user experience. Users can query the service to get a list of all possible operations on a service and decide what they have to do.
+
+**How to fix the violation**: Add an operations API endpoint (if not already present) and add details regarding this endpoint in the corresponding OpenAPI document. Examples can be found for most RPs in this repo.
+
+Links: [Index](#index) | [Error vs. Warning](#error-vs-warning) | [Automated Rules](#automated-rules) | [RPC](#rpc-violations): [Errors](#rpc-errors) or [Warnings](#rpc-warnings) | [SDK](#sdk-violations): [Errors](#sdk-errors) or [Warnings](#sdk-warnings)
+
+### <a name="R2015" />R2015 ParameterNotDefinedInGlobalParameters
+**Output Message**: Parameter "{0}" is referenced but not defined in the global parameters section of Service Definition
+
+**Description**: Per ARM guidelines, if `subscriptionId` is used anywhere as a path parameter, it must always be defined as global parameter. `api-version` is almost always an input parameter in any ARM spec and must also be defined as a global parameter.
+
+**Why the rule is important**: To reduce duplication, maintain consistent structure in ARM specifications, and ensure `subscriptionId` and `api-version` are created as client properties in the generated code.
```

Comment:
I would include a section "Impact on generated code" where you put the last part of this sentence.